src/OpenProtocolInterpreter/Tool/Mid0043.cs

```diff
@@ -1,18 +1,74 @@
-﻿namespace OpenProtocolInterpreter.Tool
+﻿using OpenProtocolInterpreter.Converters;
+using System.Collections.Generic;
+
+namespace OpenProtocolInterpreter.Tool
 {
     /// <summary>
     /// Enable tool
     /// <para>Message sent by: Integrator</para>
-    /// <para>Answer: <see cref="Communication.Mid0005"/> Command accepted</para>
+    /// <para>Answer: <see cref="Communication.Mid0005"/>Command accepted</para>
+    /// <para>Answer: <see cref="Communication.Mid0004"/>Command error, with code:
+    /// <list type="table">
+    /// <listheader>
+    /// <term>Code</term>
+    /// <description>Error description</description>
+    /// </listheader>
+    /// <item>
+    /// <term>29</term>
+    /// <description>Tool does not exist</description>
+    /// </item>
+    /// </list>
+    /// </para>
     /// </summary>
     public class Mid0043 : Mid, ITool, IIntegrator
     {
-        private const int LAST_REVISION = 1;
+        private readonly IValueConverter<int> _intConverter;
+        private const int LAST_REVISION = 2;
         public const int MID = 43;
 
+        public int ToolNumber
+        {
+            get => GetField(2, (int)DataFields.TOOL_NUMBER).GetValue(_intConverter.Convert);
+            private set => GetField(2, (int)DataFields.TOOL_NUMBER).SetValue(_intConverter.Convert, value);
+        }
+
         public Mid0043() : base(MID, LAST_REVISION)
```

Comment:
change default constructor to call this(LAST_REVISION), otherwise it will result in _intConverter to be null.